src/actions/formatWithTag.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable import/prefer-default-export */
+import FormattingCommand from '../@types/FormattingCommand'
+import Thunk from '../@types/Thunk'
+import * as selection from '../device/selection'
+import pathToThought from '../selectors/pathToThought'
+import simplifyPath from '../selectors/simplifyPath'
+import suppressFocusStore from '../stores/suppressFocus'
+import getCommandState from '../util/getCommandState'
+import strip from '../util/strip'
+import { editThoughtActionCreator as editThought } from './editThought'
+
+/** Format the browser selection or cursor thought with the provided tag. */
+export const formatWithTagActionCreator =
+  (tag: FormattingCommand): Thunk =>
+  (dispatch, getState) => {
+    const state = getState()
+    if (!state.cursor) return
+    const thought = pathToThought(state, state.cursor)
+    const simplePath = simplifyPath(state, state.cursor)
+    suppressFocusStore.update(true)
+
+    const tagRegExp = new RegExp(`<${tag}[^>]*>|<\/${tag}>`, 'g')
+
+    const thoughtContentEditable = document.querySelector(`[aria-label="editable-${thought.id}"]`)
+    if (!thoughtContentEditable) return
```

Comment:
Isn't `thoughtContentEditable` guaranteed to exist if `state.cursor` is non-null? Or is there a situation I'm not thinking of?

---

Technically it's possible to have two elements with the same `aria-label="editable-${thought.id}"` rendered if the context view is open.

Instead, you can select the cursor editable as follows:

```ts
document.querySelector(`[data-editing="true"] [data-editable="true"]`)
```

Still, it would be better to avoid DOM access in the action creator if possible.

Reply:
Good catch! This was inadvertently included from a previous implementation attempt and is no longer needed. The `thoughtContentEditable` was being used previously, so the null check and early return were so that the value could be used. Since we aren't using `thoughtContentEditable` anymore, I have removed it here: 78285031f6b1651e4f8c334aa73a4c7a9f23d746